syntax = "proto3";

service StorageService {
  rpc StoreChunk(StoreChunkRequest) returns (StoreChunkResponse);
  rpc RetrieveChunk(RetrieveChunkRequest) returns (RetrieveChunkResponse);
}

message StoreChunkRequest {
  string chunk_hash = 1;
  bytes data = 2;
}

message StoreChunkResponse { bool success = 1; }

message RetrieveChunkRequest {
  string chunk_hash = 1;
}

message RetrieveChunkResponse {
  bytes data = 1;
}
